Core/Core/API/APIPostPolicyRequestable.swift

```diff
@@ -0,0 +1,84 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public enum PostGradePolicy: String, CaseIterable {
+    case everyone, graded
+}
+
+public struct PostAssignmentGradesPostPolicyRequest: GraphqlRequestable {
+    public typealias Response = APINoContent
+
+    public let postPolicy: PostGradePolicy
+    public let assignmentID: String
+
+    public init(assignmentID: String, postPolicy: PostGradePolicy) {
+        self.postPolicy = postPolicy
+        self.assignmentID = assignmentID
+    }
+
+    public var query: String? {
+        return """
+        mutation PostAssignmentGrades
+            {
+                postAssignmentGrades(input: {assignmentId: "\(assignmentID)", gradedOnly: \(postPolicy == .graded)})
+                {
+                    assignment { id }
+                }
+            }
+        """
+    }
+}
+
+public struct GetAssignmentPostPolicyInfoRequest: GraphqlRequestable {
+    public typealias Response = APIPostPolicyInfo
+
+    public let courseID: String
+    public let assignmentID: String
+
+    public init(courseID: String, assignmentID: String) {
+        self.courseID = courseID
+        self.assignmentID = assignmentID
+    }
+
+    public var query: String? {
+        return """
+        {
+            course(id: "\(courseID)") {
+                sectionsConnection {
+                  nodes {
```

Comment:
If you want a slightly better looking response you can rename the attributes for the response by doing `sections: nodes`. That way when you access values you can do `course.sectionsConnection.sections` instead of `course.sectionsConnections.nodes`.  Small thing and I'm not asking for anything to change. Just if you want to.

Reply:
oh that's right, you can alias.  would that be `sections: sectionsConnection` or vice versa?  Great suggestion.

Reply:
oh you're suggesting for nodes, got it.  I really don't like the sectionsConnection. hahah, I would change that one first

Reply:
It goes `alias: sectionsConnection`. You can put it on either the `sectionsConnection` or the `nodes`. Just depends on how you want to access the data. `course.sections.nodes`, or `course.sectionsConnection.sections`. Or you could even alias both `course.sectionsList.sections`.

Reply:
The other option on connections is to use `edges` instead of `nodes`. Which would look like
```
sections: sectionsConnection {
  edges: {
    item: node {
      ...
    }
  }
}
```
Which you would then access as `sections.edges[0].item`. But I kind of like nodes better. Didn't know it existed until now.

Reply:
thanks for the great suggestion, have updated